### PR TITLE
VR-5548 endpoint update resources

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: client/verta/verta/deployment/resources.py [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
 
 
 class CpuMilli(_Resource):
-    raise NotImplementedError
+    def __init__(self, parameter):
+        try:
+            if parameter <= 0:
+                raise ValueError('Wrong parameter provided')
+        except TypeError:
+            raise ValueError('Wrong parameter type provided')
```
Comment (convoliution) [inline client/verta/verta/deployment/resources.py]: Let's have a slightly more specific error message to help the user, for example:
https://github.com/VertaAI/modeldb/blob/ef91ff1/client/verta/verta/deployment/update/_strategies.py#L46-L50

File: client/verta/verta/_deployment/endpoint.py [hunk 2]
```diff
@@ -213,3 +213,9 @@ def get_access_token(self):
         if len(tokens) == 0:
             return None
         return tokens[0]['creator_request']['value']
+
+    def form_update_body(self, resources, strategy, build_id):
```
Comment (convoliution) [inline client/verta/verta/_deployment/endpoint.py]: This seems like something that should be considered private.
```suggestion
    def _form_update_body(self, resources, strategy, build_id):
```

File: client/verta/verta/_deployment/endpoint.py [hunk 3]
```diff
@@ -213,3 +213,9 @@ def get_access_token(self):
...
+        update_body = strategy._as_build_update_req_body(build_id)
+        update_body["resources"] = reduce(lambda resource_a, resource_b: {**resource_a, **resource_b}, map(lambda resource: resource.to_dict(), resources))
```
Comment (convoliution) [inline client/verta/verta/_deployment/endpoint.py]:
```python
{**resource_a, **resource_b}
```
This syntax is not supported in Python 2. I think we will need a different way to merge dicts.

**Review by convoliution — CHANGES_REQUESTED** [review]

File: client/verta/verta/deployment/resources.py (hunk 1 above)
Comment (evgeniilezhnin) [reply client/verta/verta/deployment/resources.py]: replaced

File: client/verta/verta/_deployment/endpoint.py (hunk 2 above)
Comment (evgeniilezhnin) [reply client/verta/verta/_deployment/endpoint.py]: made private

File: client/verta/verta/_deployment/endpoint.py (hunk 3 above)
Comment (evgeniilezhnin) [reply client/verta/verta/_deployment/endpoint.py]: replaced to python2 syntax

**Review by convoliution — APPROVED** [review]
Thanks, Evgenii!

**Pull request merged**